deployer/server.go

```diff
@@ -0,0 +1,406 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-sql"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/pq"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/jackc/pgx"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/controller/client"
+	"github.com/flynn/flynn/deployer/strategies"
+	"github.com/flynn/flynn/deployer/types"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/postgres"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+var q *que.Client
+var db *postgres.DB
+var client *controller.Client
+
+var ErrNotFound = errors.New("deployer: resource not found")
+
+type deployID struct {
+	ID string
+}
+
+func main() {
+	var err error
+	client, err = controller.NewClient("", os.Getenv("CONTROLLER_AUTH_KEY"))
+	if err != nil {
+		log.Fatalln("Unable to connect to controller:", err)
+	}
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "4000"
+	}
+
+	addr := ":" + port
+	if err := discoverd.Register("flynn-deployer", addr); err != nil {
+		log.Fatal(err)
+	}
+
+	postgres.Wait("")
+	db, err = postgres.Open("", "")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := migrateDB(db.DB); err != nil {
+		log.Fatal(err)
+	}
+
+	pgxcfg, err := pgx.ParseURI(fmt.Sprintf("http://%s:%s@%s/%s", os.Getenv("PGUSER"), os.Getenv("PGPASSWORD"), db.Addr(), os.Getenv("PGDATABASE")))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	pgxpool, err := pgx.NewConnPool(pgx.ConnPoolConfig{
+		ConnConfig:   pgxcfg,
+		AfterConnect: que.PrepareStatements,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer pgxpool.Close()
+
+	q = que.NewClient(pgxpool)
+	wm := que.WorkMap{
+		"Deployment": handleJob,
+	}
+
+	// TODO start new worker on error
+	workers := que.NewWorkerPool(q, wm, 10)
+	go workers.Start()
+	defer workers.Shutdown()
+
+	router := httprouter.New()
+	router.POST("/deployments", createDeployment)
+	router.GET("/deployments/:deployment_id/events", streamDeploymentEvents)
+
+	log.Println("Listening for HTTP requests on", addr)
```

Comment:
We have plans to standardise logging using [log15](https://github.com/inconshreveable/log15) (#34), so we should probably kick that off by using it in the deployer